instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/InstrumenterBuilder.java

```diff
@@ -146,25 +141,26 @@
   /**
    * Enables suppression based on client instrumentation type.
    *
-   * <p><strong>When enabled, suppresses nested spans depending on their {@link SpanKind} and
-   * type</strong>.
+   * <p><strong>When enabled, suppresses nested spans depending on their {@link SpanKind} and {@link
+   * SpanKey}</strong>.
    *
    * <ul>
-   *   <li>CLIENT and PRODUCER nested spans are suppressed based on their type (HTTP, RPC, DB,
-   *       MESSAGING) i.e. if span with the same type is on the context, new span of this type will
-   *       not be started.
+   *   <li>CLIENT nested spans are suppressed based on their {@link SpanKey} (HTTP, RPC, DB) i.e. if
+   *       span marked with the same {@link SpanKey} is present in the parent context object, new
+   *       span of the same {@link SpanKey} will not be started.
    * </ul>
    *
    * <p><strong>When disabled:</strong>
    *
    * <ul>
-   *   <li>CLIENT and PRODUCER nested spans are always suppressed
+   *   <li>CLIENT nested spans are always suppressed
    * </ul>
    *
    * <p><strong>In both cases:</strong>
    *
    * <ul>
-   *   <li>SERVER and CONSUMER nested spans are always suppressed
+   *   <li>SERVER and PRODUCER nested spans are always suppressed
+   *   <li>CONSUMER spans are suppressed depending on their operation: receive or process
    *   <li>INTERNAL spans are never suppressed
    * </ul>
```

Comment:
I haven't read the full PR, but I sense some inconsistencies in this documentation.

First, `PRODUCER` is equivalent to the `CLIENT`, why are they handled differently? E.g. if there is a messaging client that uses http protocol underneath, shouldn't we suppress http client by default and not suppress when using suppression by type?

Second, this javadoc talks about `SpanKey` in case of `CLIENT` spans but uses different language for `CONSUMER` spans, although they are both implemented using `SpanKey`.

Third, do we have to reflect this somehow in some user-facing documentation?

Reply:
> First, `PRODUCER` is equivalent to the `CLIENT`, why are they handled differently?

Is it really? A `PRODUCER` puts a message with id=123 into queue with name=abc, while a `CLIENT` sends a HTTP message to URL http://... - they have different semantics, so should `PRODUCER` spans really suppress the `CLIENT` s underneath?
(Also, we've never suppressed anything related to `PRODUCER` spans in the agent)

> Second, this javadoc talks about `SpanKey` in case of `CLIENT` spans but uses different language for `CONSUMER` spans, although they are both implemented using `SpanKey`.

Good point, will fix that 👍 

> Third, do we have to reflect this somehow in some user-facing documentation?

We probably should - the various suppression behaviors exhibited by the agent definitely deserve their own documentation page.

Reply:
`PRODUCER` is equivalent to `CLIENT` in the same sense as `SERVER` is equivalent to `CONSUMER`, the only difference between them in async vs sync.

Reply:
TIL https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#span-kind 😅 

Ideally we should avoid suppressing based on the span kind and use the instrumentation/semantic type instead - it'd probably be a lot easier if we changed the default behavior from "suppress all nested clients" to the "suppress per type" one.